t/07-verify-trusted-cert.t

```diff
@@ -0,0 +1,133 @@
+use strict;
+use warnings;
+use Crypt::OpenSSL::Verify;
+use Crypt::OpenSSL::X509;
+use IO::Socket::SSL;
+use Net::SSLeay;
+use Data::Dumper;
+use File::Slurp qw{ write_file };
+use Test::More;
+
+my $openssl_version = `openssl version`;
+$openssl_version =~ /OpenSSL ([\d\.]+)/;
+$openssl_version = $1;
+
+my %chain = ();
+my $inter_cnt = 1;
+
+my $server_name = 'www.google.com';
+
+sub verify_callback {
+    my $cert = $_[4];
+    my $subject = Net::SSLeay::X509_NAME_oneline(Net::SSLeay::X509_get_subject_name($cert));
+    my $issuer  = Net::SSLeay::X509_NAME_oneline(Net::SSLeay::X509_get_issuer_name($cert));
+
+    $subject =~ /CN=(.*$)/;
+    $subject = $1;
+    $issuer =~ /CN=(.*$)/;
+    $issuer = $1;
+    if ( $subject eq $server_name ) {
+        $chain{'server'} = { name => $subject, x509 => Net::SSLeay::PEM_get_string_X509($cert), }; 
+    } elsif ( $subject ne $issuer ) {
+        my $int = 'intermediate' . $inter_cnt;
+        $chain{'intermediates'} = $inter_cnt;
+        $chain{$int} = { 'name' => $subject, 'x509' => Net::SSLeay::PEM_get_string_X509($cert), }; 
+        $inter_cnt++;
+    } elsif ( $subject eq $issuer ) {
+        $chain{'root'} = { 'name' => $subject, 'x509' => Net::SSLeay::PEM_get_string_X509($cert), };
+    }
+    return 1;
+}
+
+sub get_cert_chain {
+    my $peer = shift;
+    IO::Socket::SSL->new(
+        PeerHost => $peer . ":443",
+        SSL_verify_callback => \&verify_callback
+    ) or die $SSL_ERROR||$!;
+}
+
+get_cert_chain($server_name);
+
+my $cert = $chain{'server'}{'x509'};
+
+my $intermediate = '';
+for ( my $i = 1; $i <= $chain{intermediates}; $i++ ) {
+    $intermediate = $intermediate . $chain{"intermediate$i"}{'x509'} ."\n";
+}
+
+write_file('intermediate.pem', $intermediate);
+write_file('cert.pem', $cert);
+
+my $ret;
+
+SKIP: {
+    skip "openSSL not installed", 1 unless `which openssl`;
```

Comment:
I would do this in a skip_all like situation. 

```
plan skip_all => "Test::DistManifest required for testing the manifest"
  unless `which openssl`
```

in the top of the testfile